tests/ui/specialization/min_specialization/bad-const-wf-doesnt-specialize.rs

```diff
@@ -6,7 +6,7 @@
 struct S<const L: usize>;
 
 impl<const N: i32> Copy for S<N> {}
-//~^ ERROR the constant `N` is not of type `usize`
```

Comment:
This error disappeared, which is kinda bad imo. It would also be solved by moving the type check before overlap checking